Remotion/Web/UnitTests/Core/WebStringTest.cs

```diff
@@ -0,0 +1,210 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Web.UI;
+using NUnit.Framework;
+
+namespace Remotion.Web.UnitTests.Core
+{
+  [TestFixture]
+  public class WebStringTest
+  {
+    [Test]
+    public void CreateFromRawAttribute ()
+    {
+      var stringValue = "this & that";
+      var webString = WebString.CreateFromRawAttribute (stringValue);
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Attribute));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromRawAttribute_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromRawAttribute (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void CreateFromRawHtml ()
+    {
+      var stringValue = "<a>test</a>";
+      var webString = WebString.CreateFromRawHtml (stringValue);
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Html));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromRawHtml_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromRawHtml (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void CreateFromString ()
+    {
+      var stringValue = "My <b>test</a> string";
+      var webString = WebString.CreateFromString (stringValue);
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void CreateFromString_WithNullValue_ThrowsArgumentNullException ()
+    {
+      Assert.That (() => WebString.CreateFromString (null), Throws.ArgumentNullException);
+    }
+
+    [Test]
+    public void ImplicitConversionToString ()
+    {
+      var stringValue = "My <b>test</a> string";
+      WebString webString = stringValue;
+
+      Assert.That (webString.Type, Is.EqualTo (WebStringType.Text));
+      Assert.That (webString.GetRawValue(), Is.EqualTo (stringValue));
+    }
+
+    [Test]
+    public void IsEmpty ()
+    {
+      Assert.That (new WebString().IsEmpty, Is.True);
+      Assert.That (((WebString)"").IsEmpty, Is.True);
+      Assert.That (((WebString)" ").IsEmpty, Is.False);
+      Assert.That (((WebString)"test").IsEmpty, Is.False);
+    }
+
+    [Test]
+    public void Write_Text ()
```

Comment:
Please add result qualifier to test methods
Write_Text_EncodesOutput / ..._DoesNotEncodeOutput